### PR TITLE
[WIP] Version of #9720 without the dependency on #9728

### PR DESCRIPTION
**Pull request by justinsb**
#9720 is really important (IMO), #9728 is more of a nicety.  But #9720 depends on #9728.  This PR attempts to break that dependency.

### CODE REVIEW

**Comment by dchen1107** [comment]
cc/ @roberthbailey

**Comment by roberthbailey** [comment]
Seems reasonable. @cjcullen should take a look as well before this merges.

**Comment by cjcullen** [comment]
LGTM. I think this level of abstraction is much better for extending this to AWS (and probably elsewhere too), and it appears to keep GCE's functionality the same.

File: pkg/cloudprovider/aws/aws_routes.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+)
+
+func (s *AWSCloud) findRouteTable(clusterName string) (*ec2.RouteTable, error) {
+	request := &ec2.DescribeRouteTablesInput{}
+	filters := []*ec2.Filter{}
+	// This should be unnecessary (we already filter on TagNameKubernetesCluster,
+	// and something is broken if cluster name doesn't match, but anyway...
+	// TODO: All clouds should be cluster-aware by default
+	filters = append(filters, newEc2Filter("tag:"+TagNameKubernetesCluster, clusterName))
+	request.Filters = s.addFilters(filters)
```
Comment (cjcullen) [inline pkg/cloudprovider/aws/aws_routes.go]: You could avoid the empty declarations above and just have:
```
filters := []*ec2.Filter{{newEc2Filter("tag:"+TagNameKubernetesCluster, clusterName))}}
request := &ec2.DescribeRouteTablesInput{ Filters: s.addFilters(filters) }
```

**Comment by justinsb** [comment]
Closing as now a duplicate of #9720, now that that has merged

**Pull request closed without merging**